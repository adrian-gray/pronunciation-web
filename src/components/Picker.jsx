import React, { Component } from 'react'

import {
  withStyles
} from '@material-ui/core'

const styles = (theme) => ({
  clickable: {
    cursor: 'pointer'
  },
  correct: {
    backgroundColor: '#CCFFCC'
  },
  incorrect: {
    backgroundColor: '#FFCCCC'
  }
})

class Picker extends Component {
  constructor (props) {
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (e) {
    this.props.handleClick(this.props.index)
  }

  render () {
    return (
      <span onClick={this.handleClick}>
        {`( ${this.props.options[this.props.selected]} )`}
      </span>
    )
  }
}

export default withStyles(styles)(Picker)
