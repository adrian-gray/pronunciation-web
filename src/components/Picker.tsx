import React from "react";

export default props => {
  const handleClick = e => {
    props.handleClick(props.index);
  };

  return (
    <span className={`clickable ${props.colour}`} onClick={handleClick}>
      {`${props.options[props.selected]}`}
    </span>
  );
};
