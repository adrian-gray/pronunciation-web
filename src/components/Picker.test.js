import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Picker from "./Picker";

afterEach(cleanup);

test("Picker", async () => {
  const props = {};

  const container = shallow(<Picker {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
