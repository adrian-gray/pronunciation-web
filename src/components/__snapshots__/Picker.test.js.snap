// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker 1`] = `
<div>
  <span
    class="clickable red"
  >
    dog
  </span>
</div>
`;
